specfile/specfile.py

```diff
@@ -104,3 +115,72 @@ def tags(self, section: str = "package") -> Iterator[Tags]:
                 yield tags
             finally:
                 raw_section.data = tags.reassemble()
+
+    @contextlib.contextmanager
+    def changelog(self) -> Iterator[Optional[Changelog]]:
+        """
+        Context manager for accessing changelog.
+
+        Yields:
+            Spec file changelog as `Changelog` object or None if there is no %changelog section.
+        """
+        with self.sections() as sections:
+            try:
+                section = sections.changelog
+            except AttributeError:
+                yield None
+            else:
+                changelog = Changelog.parse(section)
+                try:
+                    yield changelog
+                finally:
+                    section.data = changelog.reassemble()
+
+    def add_changelog_entry(
+        self,
+        entry: Union[str, List[str]],
+        author: Optional[str] = None,
+        email: Optional[str] = None,
+        timestamp: Optional[datetime.datetime] = None,
+        legacy: Optional[bool] = None,
+    ) -> None:
+        """
+        Adds a new %changelog entry. Does nothing if there is no %changelog section.
+
+        If not specified, author and e-mail will be determined using rpmdev-packager, if available.
+        Timestamp, if not set, will be set to current time (in local timezone).
+
+        Args:
+            entry: Entry text or list of entry lines.
+            author: Author of the entry.
+            email: E-mail of the author.
+            timestamp: Timestamp of the entry.
+            legacy: Use legacy (date only) timestamp format.
+        """
+
+        def get_author():
+            if author is not None:
+                return author + (f" <{email}>" if email else "")
+            try:
+                result = subprocess.check_output("rpmdev-packager").decode().strip()
+            except subprocess.CalledProcessError as e:
+                raise SpecfileException("Failed to auto-detect author") from e
+            else:
+                return result
+
+        with self.changelog() as changelog:
```

Comment:
:+1: for using a context manager --  both saving and manipulation looks really clear.